lib/utils/utils.go

```diff
@@ -145,52 +148,133 @@ func KeystoreFilepaths(basepath string) ([]string, error) {
+func GetGssmrGenesisRawPathTest(t *testing.T) string {
+	t.Helper()
+	path, err := GetGssmrGenesisRawPath()
+	require.NoError(t, err)
+	return path
+}
+
 // GetGssmrGenesisRawPath gets the gssmr raw genesis path
-func GetGssmrGenesisRawPath() string {
-	path1 := "../chain/gssmr/genesis.json"
-	path2 := "../../chain/gssmr/genesis.json"
+// and returns an error if it cannot find it.
+func GetGssmrGenesisRawPath() (path string, err error) {
+	rootPath, err := GetProjectRootPath()
+	if err != nil {
+		return "", err
+	}
+	return filepath.Join(rootPath, "./chain/gssmr/genesis.json"), nil
+}
 
-	var fp string
+// GetGssmrGenesisPathTest gets the gssmr genesis path
+// and fails the test if it cannot find it.
+func GetGssmrGenesisPathTest(t *testing.T) string {
+	t.Helper()
+	path, err := GetGssmrGenesisPath()
+	require.NoError(t, err)
+	return path
+}
+
+// GetDevGenesisPath gets the dev genesis path
+func GetDevGenesisPath(t *testing.T) string {
+	return filepath.Join(GetProjectRootPathTest(t), "./chain/dev/genesis.json")
+}
 
-	if PathExists(path1) {
-		fp, _ = filepath.Abs(path1)
-	} else if PathExists(path2) {
-		fp, _ = filepath.Abs(path2)
+// GetGssmrGenesisPath gets the gssmr genesis path
+// and returns an error if it cannot find it.
+func GetGssmrGenesisPath() (path string, err error) {
+	rootPath, err := GetProjectRootPath()
+	if err != nil {
+		return "", err
 	}
+	return filepath.Join(rootPath, "./chain/gssmr/genesis-spec.json"), nil
+}
 
-	return fp
+// GetKusamaGenesisPath gets the Kusama genesis path
+func GetKusamaGenesisPath(t *testing.T) string {
+	return filepath.Join(GetProjectRootPathTest(t), "./chain/kusama/genesis.json")
 }
 
-// GetGssmrGenesisPath gets the gssmr human-readable genesis path
-func GetGssmrGenesisPath() string {
-	path1 := "../chain/gssmr/genesis-spec.json"
-	path2 := "../../chain/gssmr/genesis-spec.json"
+// GetPolkadotGenesisPath gets the Polkadot genesis path
+func GetPolkadotGenesisPath(t *testing.T) string {
+	return filepath.Join(GetProjectRootPathTest(t), "./chain/polkadot/genesis.json")
+}
 
-	var fp string
+// GetProjectRootPathTest finds the root of the project where `go.mod` is
+// and returns it as an absolute path. It fails the test if it's not found.
+func GetProjectRootPathTest(t *testing.T) (rootPath string) {
+	t.Helper()
+	rootPath, err := GetProjectRootPath()
+	require.NoError(t, err)
+	return rootPath
+}
 
-	if PathExists(path1) {
-		fp, _ = filepath.Abs(path1)
-	} else if PathExists(path2) {
-		fp, _ = filepath.Abs(path2)
-	}
+var (
+	ErrFindProjectRoot = errors.New("cannot find project root")
+)
 
-	return fp
-}
+// GetProjectRootPath finds the root of the project where `go.mod` is
+// and returns it as an absolute path.
+func GetProjectRootPath() (rootPath string, err error) {
+	_, fullpath, _, _ := runtime.Caller(0)
+	rootPath = path.Dir(fullpath)
+
+	const directoryToFind = "chain"
+	const subPathsToFind = "dev,gssmr,kusama,polkadot"
+
+	subPaths := strings.Split(subPathsToFind, ",")
```

Comment:
```suggestion
	subPaths := [...]string{"dev", "gssmr", "kusama", "polkadot"}
```
Then `subPaths` will be a fixed size array

Reply:
😢 I need constant immutable arrays so bad. I'm sure Rust has that.

I like your suggestion but since it doesn't prevent mutation of the array either, and especially that I got graced by the CI fully passing, I'll just merge it as is.